### PR TITLE
Test PR changes in the CI

### PR DESCRIPTION
**Pull request by v1v**
### What

First attempt to validate PR changes within the CI.

### Why

At the moment the `.ci/Jenkinsfile` and `.ci/its.groovy` where just validating releases. 

### Implementation details

- Initial support for `linux` then we can figure out the `windows` support.
- Use the CI workers that run in GCP, I assume the scripts are cloud platform agnostic. To clarify

### Tasks

- [x] Can we run the shell steps in a different cloud provider?
- [x] Location for the scripts to be copied to

### CODE REVIEW

**Comment by narph** [comment]
location for linux certs LINUX_CERT_PATH="/var/lib/waagent"


Installation on VM should be:
```
create `Elastic.ElasticAgent.linux-1` folder in /var/lib/waagent location which should contain

  - config folder
           - 0.settings 
           ex. file 
         {
   "runtimeSettings":[
      {
         "handlerSettings":{
            "protectedSettings":"....",
            "publicSettings":{
               "cloudId":"....",
               "username":"...."
            },
            "protectedSettingsCertThumbprint":"..."
         }
      }
   ]
}
  - status folder (empty)
  - install.sh (from repo)
  - enable.sh (from repo)
  - disable.sh (from repo)
  - helper.sh (from repo)
  - newconfig.sh (from repo)
  - update.sh (from repo)
  - uninstall.sh (from repo)
  - HandlerManifest.json (from repo)
  
  - HandlerEnvironment.json
    Ex. file:

[
   {
      "handlerEnvironment":{
         "heartbeatFile":"/var/lib/waagent/Elastic.ElasticAgent.linux-1/heartbeat.log",
         "logFolder":"/var/log/azure/Elastic.ElasticAgent.linux",
         "configFolder":"/var/lib/waagent/Elastic.ElasticAgent.linux-1/config",
         "eventsFolder":"/var/log/azure/Elastic.ElasticAgent.linux/events",
         "statusFolder":"/var/lib/waagent/Elastic.ElasticAgent.linux-1/status",
         "eventsFolder_preview":"/var/log/azure/Elastic.ElasticAgent.linux/events"
      },
      "version":1.0,
      "name":"Elastic.ElasticAgent.linux"
   }
]
```

location for linux certs LINUX_CERT_PATH="/var/lib/waagent"
At the same level as `Elastic.ElasticAgent.linux-1` we should have 
$THUMBPRINT.crt"
$THUMBPRINT.prv"

File: .ci/Makefile [hunk 1]
```diff
@@ -9,6 +9,20 @@ DOCKER_IMAGE=docker.elastic.co/observability-ci/azure-vm-tools
 help: ## Display this help text
 	@grep -E '^[a-zA-Z_-]+:.*?## .*$$' $(MAKEFILE_LIST) | sort | awk 'BEGIN {FS = ":.*?## "}; {printf "\033[36m%-20s\033[0m %s\n", $$1, $$2}'
 
+.PHONY: prepare-pr-context
+prepare-pr-context:  ## Prepare the files needed to be run within the VM context.
+	@echo "Copy src/handler/linux"
```
Comment (v1v) [inline .ci/Makefile:14]:
```suggestion
	@mkdir -p /var/lib/waagent/config
	@mkdir -p /var/lib/waagent/status
	@cp src/handler/linux/*.* /var/lib/waagent
	@chmod 700 /var/lib/waagent/*.sh
```

File: .ci/Makefile [hunk 2]
```diff
@@ -3,12 +3,35 @@ SHELL=/bin/bash -o pipefail
...
+LINUX_WAAGENT_PATH=/var/lib/waagent
+LINUX_FOLDER_NAME=Elastic.ElasticAgent.linux-1
 
 .PHONY: help
 .DEFAULT_GOAL := help
 help: ## Display this help text
 	@grep -E '^[a-zA-Z_-]+:.*?## .*$$' $(MAKEFILE_LIST) | sort | awk 'BEGIN {FS = ":.*?## "}; {printf "\033[36m%-20s\033[0m %s\n", $$1, $$2}'
 
+.PHONY: prepare-pr-context
+prepare-pr-context:  ## Prepare the files needed to be run within the VM context.
```
Comment (v1v) [inline .ci/Makefile:15]: Produces

```bash
var
└── lib
    └── waagent
        ├── Elastic.ElasticAgent.linux-1
        │   ├── HandlerEnvironment.json
        │   ├── HandlerManifest.json
        │   ├── config
        │   │   └── 0.settings
        │   ├── disable.sh
        │   ├── enable.sh
        │   ├── helper.sh
        │   ├── install.sh
        │   ├── newconfig.sh
        │   ├── status
        │   ├── uninstall.sh
        │   └── update.sh
        ├── cert.crt
        └── cert.prv

5 directories, 12 files
```

File: .ci/Makefile [hunk 3]
```diff
@@ -3,12 +3,35 @@ SHELL=/bin/bash -o pipefail
...
+	@mkdir -p $(LINUX_WAAGENT_PATH)/$(LINUX_FOLDER_NAME)/config
+	@mkdir -p $(LINUX_WAAGENT_PATH)/$(LINUX_FOLDER_NAME)/status
+	@cp ../src/handler/linux/* $(LINUX_WAAGENT_PATH)/$(LINUX_FOLDER_NAME)/
+	@chmod 700 $(LINUX_WAAGENT_PATH)/$(LINUX_FOLDER_NAME)/*.sh
+	@cp ../test/mock/HandlerEnvironment.json $(LINUX_WAAGENT_PATH)/$(LINUX_FOLDER_NAME)/
+	@cp ../test/mock/0.settings $(LINUX_WAAGENT_PATH)/$(LINUX_FOLDER_NAME)/config/
+	@cp ../test/mock/cert* $(LINUX_WAAGENT_PATH)/
```
[the remark this answers is not in the document]
Comment (narph) [reply .ci/Makefile:22]: During the install of the vm extension:
- the cert and the private key files are added in the location "/var/lib/waagent"
- the 0.settings file is created based on the format above. The public settings are straightforward, we should add the username and cloudId, for the the `protectedSettings` value we need to encrypt the following string:  `{"password": "..."}` which will contain the password.
For `protectedSettingsCertThumbprint` should be the name of the certs.
When I tested encryption I created https://github.com/elastic/azure-vm-extension/blob/master/src/handler/linux/helper.sh#L362 , maybe that helps.

But when we setup the structure I mentioned above we need to create the 0.settings file with the right values.
I will commit an example of the cert and private key soon.

**Comment by v1v** [comment]
I just closed this PR since there has not been much work on it for quite sometime time and there might be some tech changes that might affect the implementation, so I'll close it so the teams can decide what they would like to do/use in their end to avoid adding more code to be transitioned to a different system if that's the case in the future.

**Pull request closed without merging**